.github/workflows/pull-request-WAF.yml

```diff
@@ -0,0 +1,94 @@
+name: Pull Request CI - WAF Infra
+on:
+  # Triggers the workflow on push or pull request events but only for the "main" branch.
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+      - reopened
+      - ready_for_review
+      - synchronize
+    paths:
+      - 'infra-l2-waf/**'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Required for sam-validate step
+permissions:
+  id-token: write
+  contents: read
+
+env: # Only adding the variables in that are required for 
+  AWS_REGION: eu-west-2
+
+jobs:
+  run-code-check:
+    name: Code Checks
+    runs-on: ubuntu-latest
+    environment: di-ipv-cri-dev
```

Comment:
this is actually common-cri env and we are not deploying anymore in common cri env as part of our deployment. i have taken it off in main pull-request file